### PR TITLE
Create GitWorktrees.md

### PR DESCRIPTION
**Pull request by markcmiller86**
Addresses #678

### CODE REVIEW

**Comment by prwolfe** [comment]
I worked with this a bit yesterday and noticed a few items.
* each workspace creates a branch with the same name as the basename of the directory specified
* You can remove and re-add a workspace without getting a branch conflict.
* This is still marked as experimental and warning against using it for a super project is prominent.

Overall this is a nice productivity boost we should highlight. I know we linked to the docs, but I would also state the warning and caveat in our write-up as those are not obvious in the docs.

Thanks @markcmiller86

**Comment by bartlettroscoe** [comment]
> each workspace creates a branch with the same name as the basename of the directory specified

@prwolfe, actually, that is just the default.  You can name and place the directory anywhere you want with an existing branch.  See the [official docuemntation](https://git-scm.com/docs/git-worktree).

**Comment by prwolfe** [comment]
> @prwolfe, actually, that is just the default.

True, but an unexpected one.

**Comment by bartlettroscoe** [comment]
> > @prwolfe, actually, that is just the default.
>
> True, but an unexpected one.

And hence my point that git worktrees adds more things to learn about git without really giving you any significant new functionality (and breaking some other processes perhaps).  Personally, I am not sold.

**Comment by bartlettroscoe** [comment]
@markcmiller86, I fixed some typos and made a few small suggested changes in the new branch and PR #691 against the branch in this PR.  Merge and/or modify if you like.

**Comment by bartlettroscoe** [comment]
@markcmiller86, FYI, I added "Addresses #678" [above](https://github.com/betterscientificsoftware/bssw.io/pull/688#issue-562926755) to provide bidirectional linkage between the motivating issue #678 and this PR #688 that addresses that issue.

**Comment by markcmiller86** [comment]
@rinku I've incorporated changes from @bartlettroscoe and @prwolfe. This is ready.

**Review by bartlettroscoe — APPROVED** [review]
@markcmiller86, the updates look great to me.  I think this is ready to publish.

**Comment by bartlettroscoe** [comment]
@bernhold, @rinkug, why don't we see this article at:

* https://preview.bssw.io/items?topic=revision-control

?

**Comment by rinkug** [comment]
> @bernhold, @rinkug, why don't we see this article at:
> * https://preview.bssw.io/items?topic=revision-control

Because the file has no metadata in it?

**Comment by bartlettroscoe** [comment]
>Because the file has no metadata in it?

@markcmiller86, @rinkug, do we have a checklist for CC contributions that reminds us of this stuff?  Like:

* [ ] Finish review of the content
* [ ] Add meta-data (what state does it need to be in?)
* [ ] Rebuild preview.bssw.io and view the new content there and how it fits into the site
* [ ] Make any final changes needed verify on preview.bssw.io
* [ ] Merge PR
* [ ] Verify new new contribution shows up on bssw.io

?

We should put the checklist in the first comment of every new PR (which we can do automatically with a PR template).

**Comment by markcmiller86** [comment]
> Because the file has no metadata in it?

I was thinking this was a step someone else took care of. Sorry if I was wrong.

**Comment by markcmiller86** [comment]
> @markcmiller86, @rinkug, do we have a checklist for CC contributions that reminds us of this stuff? Like:

Great idea. Can you submit a PR? Or, did you want me to?

**Comment by markcmiller86** [comment]
I've added metadata section. Filled in categories and topics as best I could. Was surprised that `productivity` appears only under the `Skills` category and in that context it is just `personal productivity`. I would have considered this CC to be about `development productivity` but we don't have such a topic area. Shouldn't we?

**Comment by bartlettroscoe** [comment]
> > @markcmiller86, @rinkug, do we have a checklist for CC contributions that reminds us of this stuff? Like:
> 
> Great idea. Can you submit a PR? Or, did you want me to?

@markcmiller86, I have not created a template before but I can figure it out if you want.

But before we create a template, do we have permission from @rinkug to do so?

**Comment by rinkug** [comment]
> > Because the file has no metadata in it?
> 
> @markcmiller86, @rinkug, do we have a checklist for CC contributions that reminds us of this stuff? Like:
> 
> * [ ]  Finish review of the content
> * [ ]  Add meta-data (what state does it need to be in?)
> * [ ]  Rebuild preview.bssw.io and view the new content there and how it fits into the site
> * [ ]  Make any final changes needed verify on preview.bssw.io
> * [ ]  Merge PR
> * [ ]  Verify new new contribution shows up on bssw.io

> We should put the checklist in the first comment of every new PR (which we can do automatically with a PR template).


@bartlettroscoe, @markcmiller86: Some of these items are mentioned in the [EB docs website - content publishing](https://betterscientificsoftware.github.io/bssw.io/bssw_content_publishing.html) section but it was written before the preview branch. Maybe you add these things to "Item Review" section, through a PR?

**Comment by markcmiller86** [comment]
Apologies but I'm perplexed now. Is there anything further needed from me on either this article or #670? I feel like I am done with both.

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe, @markcmiller86: Some of these items are mentioned in the [EB docs website - content publishing](https://betterscientificsoftware.github.io/bssw.io/bssw_content_publishing.html) section but it was written before the preview branch. Maybe you add these things to "Item Review" section, through a PR?

@rinkug, yes the EB docs should be updated as well but I feel strongly that each PR should have a checklist to make it easy for authors and EB board members to do what needs to be done.

What does the @betterscientificsoftware/bssw-editorial-board think about putting a checklist in each PR as suggested [above](https://github.com/betterscientificsoftware/bssw.io/pull/688#issuecomment-770091127)?

**Comment by bartlettroscoe** [comment]
> What does the @betterscientificsoftware/bssw-editorial-board think about putting a checklist in each PR as suggested [above](https://github.com/betterscientificsoftware/bssw.io/pull/688#issuecomment-770091127)?

Actually, we should move this discussion into a different issue under the [BSSw Internal](https://github.com/betterscientificsoftware/bssw.io/projects/2) board.  Should I create a new one or is there one already for this we should use?

**Comment by rinkug** [comment]
> > What does the @betterscientificsoftware/bssw-editorial-board think about putting a checklist in each PR as suggested [above](https://github.com/betterscientificsoftware/bssw.io/pull/688#issuecomment-770091127)?
> 
> Actually, we should move this discussion into a different issue under the [BSSw Internal](https://github.com/betterscientificsoftware/bssw.io/projects/2) board. Should I create a new one or is there one already for this we should use?

@bartlettroscoe : Yes, this should be a new issue. (to answer your question: personally, I think, putting checklist in a PR is good idea, though (1) it will have duplication and we will have to ensure that its always consistent with EB docs, (2) technically, can we even create a template for a PR? - Lets continue discussion in the new issue, @berhold can chime in his thoughts in that issue, as well)

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe : Yes, this should be a new issue. (to answer your question: personally, I think, putting checklist in a PR is good idea, though (1) it will have duplication and we will have to ensure that its always consistent with EB docs, (2) technically, can we even create a template for a PR? - Lets continue discussion in the new issue, @berhold can chime in his thoughts in that issue, as well)

@rinkug, @markcmiller86, I created #695.  Let's continue discussion on a PR checklist there.

File: CuratedContent/GitWorktrees.md [hunk 1]
```diff
@@ -20,8 +20,8 @@ In addition, if some work is incomplete but in progress, a common practice is to
 
 However, when working on multiple branches of development *simultaneously*, there is often more involved than just
 stashing half-completed work away or switching the currently active branch. Some code may need to be re-configured
-and/or re-compiled; maybe a lot of it. This is particularly true of compiled-language projects which are sensitive
-to timestamp changes of any build tree dependencies. Resulting delays can create a productivity issue.
+and/or re-compiled; maybe a lot of it. This is particularly true for projects which are sensitive
+to timestamp changes of *any* build tree dependencies. Resulting delays can create a productivity issue.
```
Comment (bartlettroscoe) [inline CuratedContent/GitWorktrees.md:24]: Thumbs up

**Comment by markcmiller86** [comment]
@rinkug I've viewed on preview site and it looked good. I think this is read too.

**Review by rinkug — APPROVED** [review]

**Comment by bartlettroscoe** [comment]
Related to my epic SEPW-211

**Pull request merged**